app/Nova/Signature.php

```diff
@@ -0,0 +1,157 @@
+use Laravel\Nova\Fields\ID;
+use Laravel\Nova\Fields\Text;
+use Laravel\Nova\Panel;
+use Outhebox\NovaHiddenField\HiddenField as Hidden;
+
+/**
+ * A Nova resource for signatures.
+ *
+ * @property bool $electronic
+ * @property ?string $scanned_agreement
+ * @property \Carbon\Carbon $updated_at
+ */
+class Signature extends Resource
+{
+    /**
+     * The model the resource corresponds to.
+     *
+     * @var string
+     */
+    public static $model = \App\Models\Signature::class;
+
+    /**
+     * The single value that should be used to represent the resource when being displayed.
+     *
+     * @var string
+     */
+    public static $title = 'id';
+
+    /**
+     * The columns that should be searched.
+     *
+     * @var array<string>
+     */
+    public static $search = [
+    ];
+
+    /**
+     * Indicates if the resource should be globally searchable.
+     *
+     * @var bool
+     */
+    public static $globallySearchable = false;
+
+    /**
+     * The logical group associated with the resource.
+     *
+     * @var string
+     */
+    public static $group = 'Membership Agreements';
+
+    /**
+     * Get the fields displayed by the resource.
+     */
+    public function fields(Request $request): array
+    {
+        return [
+            ID::make('ID')
+                ->sortable(),
+
+            BelongsTo::make('User')
+                ->readonly(),
+
+            BelongsTo::make('Template', 'membershipAgreementTemplate', MembershipAgreementTemplate::class)
+                ->readonly(),
+
+            Boolean::make('Complete')
+                ->exceptOnForms(),
+
+            Text::make('Type', 'electronic')
+                ->resolveUsing(static function (bool $electronic): string {
+                    return $electronic ? 'Electronic' : 'Paper';
+                })
+                ->exceptOnForms(),
+
+            DateTime::make('Rendered', 'render_timestamp')
+                ->onlyOnDetail(),
+
+            ...(! $this->electronic ? [
+                new Panel(
+                    'Paper Upload',
+                    [
+                        File::make('Scanned Agreement')
+                            ->help('Upload the entire agreement as a single file.')
+                            ->disk('local')
+                            ->deletable(false)
+                            ->required()
+                            ->rules('required'),
+
+                        BelongsTo::make('Uploaded By', 'uploadedBy', User::class)
+                            ->onlyOnDetail(),
+
+                        DateTime::make('Uploaded At', 'updated_at')
+                            ->resolveUsing(function (Carbon $str): ?Carbon {
+                                return null === $this->scanned_agreement ? null : $this->updated_at;
+                            })
+                            ->onlyOnDetail(),
+                    ]
+                ),
+            ] : []),
+
+            ...($this->electronic ? [
```

Comment:
You don't need the second ternary, you can put this panel in place of the previous ternary. I don't mind this as-is though.